brave/brave-ios
# Fix #5103: Add / Switch Network Dapps Request Integration

## Summary of Changes
- Update `NetworkPicker` to use a Style struct, update `WalletPanelView` to re-use this `NetworkPicker` menu.
- Update `CustomNetworkDetails` to add a view-only mode (everything disabled)
- Integrated `PendingWebpageRequest.switchNetwork` and `PendingWebpageRequest.addNetwork` to open the `SuggestedNetworkView` modally using `WebpageRequestContainerView`.

This pull request fixes #5103

## Submitter Checklist:


## Test Plan:
Note: At time of PR creation, wallet dapps are disabled with a compiler flag. Add -DWALLET_DAPPS_ENABLED to base.xcconfig to enable dapps support.

Network Picker in wallet panel: 
1. Visit a dapp site (ex. app.uniswap.org) and tap wallet icon in URL bar
2. Unlock wallet, click network in top right of panel

Switch Network:
1. Run [eth-manual-tests](https://github.com/bbondy/eth-manual-tests) locally
2. Click `wallet_switchEthereumChain` / `wallet_switchEthereumChain2`
Note: You may have to unlock you wallet before the modal to switch network appears
3. Tapping 'View Details' should show modal of network details, Tapping 'Cancel' should remove the pending request and dismiss, tapping 'Switch networks' should switch networks, cancel pending request and dismiss

Add Network:
1. Run [eth-manual-tests](https://github.com/bbondy/eth-manual-tests) locally
2. Click `wallet_addEthereumChain`
Note: You may have to unlock you wallet before the modal to switch network appears
3. Tapping 'View Details' should show modal of network details, Tapping 'Cancel' should remove the pending request and dismiss, tapping 'Approve' should add the network, cancel pending request and dismiss

## Screenshots:
![network picker](https://user-images.githubusercontent.com/5314553/160470272-fcee2808-c778-4523-912d-da4e1fd9f545.png) | ![switch](https://user-images.githubusercontent.com/5314553/167949978-f7d82b50-9200-4a85-bcee-ade9c25e1667.png) | ![add](https://user-images.githubusercontent.com/5314553/167924531-d79259f8-180a-40b1-b140-c7532846476b.png)
--|--|--

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
